### PR TITLE
Describe authentication module

### PR DESCRIPTION
**Pull request by EugeniyBurmistrov**

### CODE REVIEW

File: docs/modules/authentication.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+```js title="auth/route_register_callback.js"
+import jwt from 'jsonwebtoken'
+import jwtSecret from './jwt_secret' // Store your JWT secret in a secure location, for example an environment variable
+import bcrypt from 'bcrypt'
+
+// A route handler accepts `req` as the first argument. The rest of the arguments depend on the used strategy.
+// The `Local` strategy adds two arguments - `username` and `password`.
+const routeRegisterCallback = async ({ resolve }, username, password) => {
+  const { data: existingUser } = await resolve.executeQuery({ // Query a read model to check if the user already exists.
+    modelName: 'read-model-name',
```
Comment (const314) [inline docs/modules/authentication.md:113]: maybe it is better to use  'users' and 'findUser' read model and resolver names, or at least comment that it should refer to a readModel where users are stored

**Review by const314 — APPROVED** [review]

File: docs/modules/authentication.md (hunk 1 above)
Comment (EugeniyBurmistrov) [reply docs/modules/authentication.md]: Changed to 'users' and 'findUser'.

File: docs/modules/authentication.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+  // Export a function that accepts runtime options specified in the application config.
+  factory: StrategyFactory, // The passport strategy factory.
+  options: {
+    // Custom compile-time options ...
+    failureRedirect: (error) =>
+      `/error?text=${encodeURIComponent(error.message)}`,
+    errorRedirect: (error) =>
+      `/error?text=${encodeURIComponent(error.message)}`,
+    usernameField: 'username',
+    passwordField: 'username',
```
[the remark this answers is not in the document]
Comment (EugeniyBurmistrov) [reply docs/modules/authentication.md]: Fixed

File: docs/modules/authentication.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+      name: 'local-strategy', // The name of a Passport authentication strategy to use.
+      createStrategy: 'auth/create_strategy.js', // A path to the strategy constructor file within the project.
+      options: { // An object that contains custom options passed to the strategy constructor.
+        strategySecretKey: injectRuntimeEnv('STRATEGY_SECRET_KEY_ENV_VARIABLE_NAME')
+      },
+      logoutRoute: { // An HTTP route for user logout.
+          path: 'logout',
+          method: 'POST'
+      }
+      routes: [ // A list of HTTP API handlers required for for the current strategy.
```
Comment (albertov05) [inline docs/modules/authentication.md:44]:
```suggestion
      routes: [ // A list of HTTP API handlers required for the current strategy.
```

**Pull request merged**